### PR TITLE
Updating mongodb types for v6

### PR DESCRIPTION
**Pull request by Grubba27**
In this PR I remove operator types that were removed in MongoDB v6.
For more information check [MongoDB Changelog](https://www.mongodb.com/docs/manual/release-notes/6.0-compatibility/#removed-operators)

### CODE REVIEW

**Review by StorytellerCZ — APPROVED** [review]

File: packages/mongo/mongo.d.ts [hunk 1]
```diff
@@ -18,22 +18,12 @@ export namespace Mongo {
...
-    $hint?: Hint;
+  export type QueryWithModifiers = {
     $maxScan?: any;
-    $max?: any;
-    $maxTimeMS?: any;
-    $min?: any;
-    $orderby?: any;
-    $returnKey?: any;
-    $showDiskLoc?: any;
     $natural?: any;
```
Comment (radekmie) [inline packages/mongo/mongo.d.ts:23]: Interestingly, the official driver no longer has typings for any these. Maybe we should remove it too?

**Review by radekmie — APPROVED** [review]

File: packages/mongo/mongo.d.ts (hunk 1 above)
Comment (radekmie) [reply packages/mongo/mongo.d.ts]: Yeah, I meant exactly that. But now we have an absurd situation, where we have a `Query` type that's an alias for driver's `Filter` and the only place we use it is our `Selector` type 😛 

I'm not sure what we should do exactly, but I think that staying as close to the driver's types and naming would be beneficial.

Comment (Grubba27) [reply packages/mongo/mongo.d.ts]: Agreed! thanks for the insights :)

Comment (Grubba27) [reply packages/mongo/mongo.d.ts]: I hope now looks good!

**Pull request merged**